crypto/schemes/enc/v1/filekey.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"golang.org/x/crypto/chacha20poly1305"
+	"golang.org/x/crypto/hkdf"
+)
+
+type fileKey struct {
+	cipher Cipher
+
+	fileKey     []byte
+	noncePrefix []byte
+
+	// HMAC key used to sign the header
+	headerKey []byte
+	// Key used to encrypt the payload
+	payloadKey []byte
+}
+
+func newFileKey(cipher Cipher) (fileKey, error) {
+	// Read 39 random bytes for the file key (256 bits) and nonce prefix (56 bits)
+	rnd := make([]byte, 39)
+	_, err := io.ReadFull(rand.Reader, rnd)
+	if err != nil {
+		return fileKey{}, fmt.Errorf("failed to generate file key: %w", err)
```

Comment:
Go-newbie question: why are you returning an empty `fileKey` instead of just returning `nil`. Same on `importFileKey`

Reply:
That method returns a non-pointer value, so cannot return `nil` there.